### ISSUE TITLE
thread 'LayoutWorker worker 2/3' panicked at 'assertion failed: marker_fragments.len() == 1', servo/components/layout/construct.rs:1175

### ISSUE THREAD

**Issue by Yoric** [servo/servo#7030]
While visiting http://rustlang.org, with b9dcd49778b8ff6d2aef0ac2420c3a30b80432c1.

Full stack:
````
thread 'LayoutWorker worker 2/3' panicked at 'assertion failed: marker_fragments.len() == 1', /Users/david/Documents/Code/servo/components/layout/construct.rs:1175
stack backtrace:
   1:        0x103f14855 - sys::backtrace::write::h2513b694e23623efOvs
   2:        0x103f17e90 - panicking::on_panic::h399650fb5f386215wWw
   3:        0x103f04f82 - rt::unwind::begin_unwind_inner::h208861166cac0468HEw
   4:        0x101980afc - rt::unwind::begin_unwind::h13324227360576903564
   5:        0x101a72df3 - construct::FlowConstructor<'a>::build_flow_for_list_item::ha39e71ebf278c0d2n3d
   6:        0x101a57bd3 - construct::FlowConstructor<'a>.PostorderNodeMutTraversal::process::h433ee1a9c780b6998fe
   7:        0x101dad80a - traversal::ConstructFlows<'a>.PostorderDomTraversal::process::h7691bb47d2df424fm8w
   8:        0x101dad40a - parallel::ParallelPostorderDomTraversal::run_parallel::h5065035172668496833
   9:        0x101dad2db - parallel::construct_flows::h00bdc71da9b1b329ats
  10:        0x101dabe18 - parallel::RecalcStyleForNode<'a>.ParallelPreorderDomTraversal::run_parallel::hf3cad5d5615bffb0dss
  11:        0x101dad1f7 - parallel::recalc_style::h2e80ed3729834a02Dss
  12:        0x101c23d30 - workqueue::WorkerThread<QueueData, WorkData>::start::h17518797412544091150
  13:        0x101c23424 - workqueue::WorkQueue<QueueData, WorkData>::new::closure.43114
  14:        0x101c2328f - task::spawn_named::closure.43107
  15:        0x101c2319e - boxed::F.FnBox<A>::call_box::h3413909981105523445
  16:        0x101bbda30 - boxed::Box<FnBox<A, Output $u3d$$u20$R$GT$$u2b$$u20$Send$u20$$u2b$$u20$$u27$a$GT$.FnOnce$LT$A$GT$::call_once::h17258177205126700357
  17:        0x101bbd4a2 - thread::Builder::spawn_inner::closure.41293
  18:        0x101bbd41e - rt::unwind::try::try_fn::h4735211398648586166
  19:        0x103f1778f - __rust_try_inner
  20:        0x103f177ca - __rust_try
  21:        0x103f12dc5 - rt::unwind::try::inner_try::h300fa716ddfd682bAAw
  22:        0x101bbd368 - rt::unwind::try::h12410211325088419976
  23:        0x101bbd1ac - thread::Builder::spawn_inner::closure.41245
  24:        0x101bbdd4d - boxed::F.FnBox<A>::call_box::h9973539156334481435
  25:        0x103f167dd - sys::thread::Thread::new::thread_start::he9d2ef50096740e9bZv
  26:     0x7fff886f1267 - _pthread_body
  27:     0x7fff886f11e4 - _pthread_start
````

**Comment by Yoric** [comment]
Cc @mbrubeck , @pcwalton .

**Comment by jdm** [comment]
See also #6913.

**Comment by mbrubeck** [comment]
Possibly triggered by #6979.  Might be fixed by #6998.  But in general it will not always be true that a list marker is a single fragment, so we should remove that assumption from this code.

**Comment by mbrubeck** [comment]
Closing as duplicate of #6913.  Also, since #6913 was reported by #6998 landed, it must not be triggered by that bug.
